### PR TITLE
Add preliminary docs to the index methods

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: python/hail/table.py [hunk 1]
```diff
@@ -1159,6 +1159,74 @@ def _show(self, n=10, width=90, truncate=None, types=True):
...
+        |     1 | dog   |
+        |     2 | dog   |
+        |     3 | dog   |
+        |     4 | mouse |
+        +-------+-------+
+
+        Notes
+        -----
+        :meth:`.Table.index` is used to expose one table's fields for use in
+        expressions involving the another table's fields. The result of the
```
Comment (jigold) [inline python/hail/table.py]: involving another table or matrix table

File: python/hail/matrixtable.py [hunk 2]
```diff
@@ -2279,11 +2318,71 @@ def index_rows(self, *exprs):
...
+        Parameters
+        ----------
+        exprs : variable-length args of :class:`.Expression`
+            Index expressions.
+
+        Notes
+        -----
+        :meth:`index_cols` is equivalent to ``cols().index``.
+
+        The type of the resulting struct is the same as the type of
```
[the remark this answers is not in the document]
Comment (tpoterba) [reply python/hail/matrixtable.py]: we're deprecating that syntax though. Do you mean `cols()[...]`?

[the remark this answers is not in the document]
Comment (jigold) [reply python/hail/matrixtable.py]: oh, I'm wrong. The bracket notation was useful though -- why are we getting rid of it? Also, it's confusing that you can use it for `index` and `index_entries`, but not the other methods.

[the remark this answers is not in the document]
Comment (tpoterba) [reply python/hail/matrixtable.py]: the `:` syntax in Python is a broadcast / copy, which is totally the opposite of how we were using it -- as a way to restrict to one dimension. A bunch of people have gotten confused by `mt[..., :]` to do a row join. I think `mt.rows[...]` is way better. (we should also make rows/cols/entries properties!)

[the remark this answers is not in the document]
Comment (jigold) [reply python/hail/matrixtable.py]: That makes sense. Can we add the note then that mt.rows[...] is another way to do the same thing? Same for cols.

File: python/hail/matrixtable.py [hunk 3]
```diff
@@ -2210,6 +2210,19 @@ def entries(self) -> Table:
         return Table(self._jvds.entriesTable())
 
     def index_globals(self) -> Expression:
+        """Return this matrix table's global variables for use in another
+        table's expression context.
+
+        Examples
+        --------
+        >>> pli_dict = dataset.index_globals().pli
+        >>> dataset_result = dataset2.annotate_rows(gene_pli = dataset2.gene.map(pli_dict.get))
```
[the remark this answers is not in the document]
Comment (tpoterba) [reply python/hail/matrixtable.py]: ah, it doesn't work here because of the default argument on `get`

[the remark this answers is not in the document]
Comment (tpoterba) [reply python/hail/matrixtable.py]: but you could totally pass in a 1-argument function here

**Pull request merged**